lib/index.js

```diff
@@ -314,12 +314,28 @@ class GalleryImageArticle extends BaseAsset {
         }
     }
 
+    // temporal_coverage here is an array of dates, sorted
+    // in ascending order. The consumer is responsible for
+    // translating this into something sensible in accordance
+    // with the user's localisation preferences per
+    // http://schema.org/temporalCoverage
+    set_temporal_coverage(value) {
+        this._temporal_coverage = value.map((v) => {
+            if (!(v instanceof Date)) {
+                return new Date(v);
+            }
+
+            return v;
+        }).sort();
+    }
+
     to_metadata() {
         const metadata = super.to_metadata();
         Object.assign(metadata, {
             'document': this._document.html(),
             'authors': [this._author],
             'published': this._date_published.toISOString(),
+            'temporalCoverage': this._temporal_coverage || []
```

Comment:
The issue here is that `this._temporal_coverage` is not mandatory and may be undefined. In that case we should set it to an empty array. I'll set the default value in the constructor nonetheless.